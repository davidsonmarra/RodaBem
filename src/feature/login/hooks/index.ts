export {default as useFormScroll} from './use-form-scroll';
