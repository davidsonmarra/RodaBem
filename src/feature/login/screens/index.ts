export {default as SignInScreen} from './sign-in';
