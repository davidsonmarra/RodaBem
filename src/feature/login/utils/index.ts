export {default as signUpDataForm} from './sign-up-data-form';
export type {DataProps} from './sign-up-data-form';
