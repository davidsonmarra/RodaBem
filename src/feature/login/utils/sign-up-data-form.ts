import {TextInputProps} from 'react-native';

export interface DataProps {
  step: number;
  title: string;
  text: string;
  name: string;
  type: string;
  placeholder?: string;
  label?: string;
  inputOptions?: TextInputProps;
}

const signUpDataForm: DataProps[] = [
  {
    step: 0,
    title: 'Bem-vindo ao aplicativo',
    text: 'Para começar, precisamos de algumas informações suas',
    name: 'name',
    type: 'text',
    placeholder: 'Nome',
    inputOptions: {
      autoCapitalize: 'words',
      autoCorrect: false,
    },
  },
  {
    step: 1,
    title: 'E-mail',
    text: 'Digite seu e-mail',
    name: 'email',
    type: 'email',
    placeholder: 'E-mail',
    inputOptions: {
      autoCapitalize: 'none',
      autoCorrect: false,
      keyboardType: 'email-address',
    },
  },
  {
    step: 2,
    title: 'Senha',
    text: 'Digite sua senha',
    name: 'password',
    type: 'password',
    placeholder: 'Senha',
    inputOptions: {
      autoCorrect: false,
      autoCapitalize: 'none',
    },
  },
  {
    step: 3,
    title: 'Confirme sua senha',
    text: 'Digite sua senha novamente',
    name: 'confirmPassword',
    type: 'password',
    placeholder: 'Confirme sua senha',
    inputOptions: {
      autoCorrect: false,
      autoCapitalize: 'none',
    },
  },
];

export default signUpDataForm;
