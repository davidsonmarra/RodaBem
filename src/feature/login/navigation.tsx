import React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {SignInScreen, SignUpScreen} from './screens';

export type RootStackParamList = {
  SignIn: undefined;
  SignUp: undefined;
};

const {Navigator, Screen} = createNativeStackNavigator<RootStackParamList>();

const Navigation = () => {
  return (
    <Navigator screenOptions={{headerShown: false}}>
      <Screen name="SignIn" component={SignInScreen} />
      <Screen name="SignUp" component={SignUpScreen} />
    </Navigator>
  );
};

export default Navigation;
