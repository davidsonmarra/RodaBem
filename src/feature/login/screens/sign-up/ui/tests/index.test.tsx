import React from 'react';
import {TouchableOpacity} from 'react-native';
import {act, create} from 'react-test-renderer';
import SignUpContainer from '..';
import {Button} from '@components';
import {signUpDataForm} from 'src/feature/login/utils';

const mockGoBack = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({goBack: mockGoBack}),
}));

const mockOnSubmit = jest.fn();
const containerInstance = ({
  control,
  onSubmit = mockOnSubmit,
  currentStep = 0,
  listRef = jest.fn(),
}: any) =>
  create(
    <SignUpContainer
      control={control}
      onSubmit={onSubmit}
      currentStep={currentStep}
      data={signUpDataForm}
      onBack={mockGoBack}
      listRef={listRef}
      isLoading={false}
      isDisabled={false}
      setIsDisabled={jest.fn()}
    />,
  );

describe('SignUpContainer', () => {
  it('should render correctly', () => {
    const tree = containerInstance({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should goBack', () => {
    const instance = containerInstance({}).root;
    const backButton = instance.findAllByType(TouchableOpacity)[0];

    act(() => {
      backButton.props.onPress();
    });
    expect(mockGoBack).toHaveBeenCalledTimes(1);
  });

  it('should call onSubmit', () => {
    const instance = containerInstance({}).root;
    const button = instance.findByType(Button);

    act(() => {
      button.props.onPress();
    });
    expect(mockOnSubmit).toHaveBeenCalledTimes(1);
  });
});
