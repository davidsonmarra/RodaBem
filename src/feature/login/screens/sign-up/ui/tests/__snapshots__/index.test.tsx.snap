// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpContainer should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F1F1F1",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="#E30B0B"
        focusable={false}
        height="24"
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={960}
        vbWidth={960}
        width="24"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4293069579,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M359-242L120-481l239-239 43 43-166 166h604v60H236l166 166-43 43z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#121212",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          {
            "flex": 1,
            "textAlign": "center",
          },
        ]
      }
    >
      Cadastrar
    </Text>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <View
      style={
        {
          "width": 24,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "paddingHorizontal": 24,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View
              style={
                {
                  "backgroundColor": "#E30B0B",
                  "flex": 1,
                  "height": 12,
                }
              }
            />
            <View
              style={
                {
                  "width": 8,
                }
              }
            />
            <View
              style={
                {
                  "backgroundColor": "#ABABAB",
                  "flex": 1,
                  "height": 12,
                }
              }
            />
            <View
              style={
                {
                  "width": 8,
                }
              }
            />
            <View
              style={
                {
                  "backgroundColor": "#ABABAB",
                  "flex": 1,
                  "height": 12,
                }
              }
            />
            <View
              style={
                {
                  "width": 8,
                }
              }
            />
            <View
              style={
                {
                  "backgroundColor": "#ABABAB",
                  "flex": 1,
                  "height": 12,
                }
              }
            />
            <View
              style={
                {
                  "width": 8,
                }
              }
            />
          </View>
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <RCTScrollView
              bounces={false}
              data={
                [
                  {
                    "inputOptions": {
                      "autoCapitalize": "words",
                      "autoCorrect": false,
                    },
                    "name": "name",
                    "placeholder": "Nome",
                    "step": 0,
                    "text": "Para começar, precisamos de algumas informações suas",
                    "title": "Bem-vindo ao aplicativo",
                    "type": "text",
                  },
                  {
                    "inputOptions": {
                      "autoCapitalize": "none",
                      "autoCorrect": false,
                      "keyboardType": "email-address",
                    },
                    "name": "email",
                    "placeholder": "E-mail",
                    "step": 1,
                    "text": "Digite seu e-mail",
                    "title": "E-mail",
                    "type": "email",
                  },
                  {
                    "inputOptions": {
                      "autoCapitalize": "none",
                      "autoCorrect": false,
                    },
                    "name": "password",
                    "placeholder": "Senha",
                    "step": 2,
                    "text": "Digite sua senha",
                    "title": "Senha",
                    "type": "password",
                  },
                  {
                    "inputOptions": {
                      "autoCapitalize": "none",
                      "autoCorrect": false,
                    },
                    "name": "confirmPassword",
                    "placeholder": "Confirme sua senha",
                    "step": 3,
                    "text": "Digite sua senha novamente",
                    "title": "Confirme sua senha",
                    "type": "password",
                  },
                ]
              }
              decelerationRate="fast"
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={true}
              keyExtractor={[Function]}
              onContentSizeChange={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              pagingEnabled={true}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEnabled={false}
              scrollEventThrottle={0.0001}
              showsHorizontalScrollIndicator={false}
              stickyHeaderIndices={[]}
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={
                    [
                      {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "width": 702,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "height": 40,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#121212",
                            "fontSize": 24,
                            "fontWeight": "bold",
                          },
                          undefined,
                        ]
                      }
                    >
                      Bem-vindo ao aplicativo
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#222222",
                            "fontSize": 16,
                            "fontWeight": "normal",
                          },
                          undefined,
                        ]
                      }
                    >
                      Para começar, precisamos de algumas informações suas
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "borderColor": "#595959",
                            "borderRadius": 10,
                            "borderWidth": 1,
                            "flexDirection": "row",
                            "height": 62,
                            "paddingHorizontal": 10,
                            "width": "100%",
                          }
                        }
                      >
                        <TextInput
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder="Nome"
                          style={
                            {
                              "flex": 1,
                            }
                          }
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={
                    [
                      {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "width": 702,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "height": 40,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#121212",
                            "fontSize": 24,
                            "fontWeight": "bold",
                          },
                          undefined,
                        ]
                      }
                    >
                      E-mail
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#222222",
                            "fontSize": 16,
                            "fontWeight": "normal",
                          },
                          undefined,
                        ]
                      }
                    >
                      Digite seu e-mail
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "borderColor": "#595959",
                            "borderRadius": 10,
                            "borderWidth": 1,
                            "flexDirection": "row",
                            "height": 62,
                            "paddingHorizontal": 10,
                            "width": "100%",
                          }
                        }
                      >
                        <TextInput
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder="E-mail"
                          style={
                            {
                              "flex": 1,
                            }
                          }
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={
                    [
                      {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "width": 702,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "height": 40,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#121212",
                            "fontSize": 24,
                            "fontWeight": "bold",
                          },
                          undefined,
                        ]
                      }
                    >
                      Senha
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#222222",
                            "fontSize": 16,
                            "fontWeight": "normal",
                          },
                          undefined,
                        ]
                      }
                    >
                      Digite sua senha
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "borderColor": "#595959",
                            "borderRadius": 10,
                            "borderWidth": 1,
                            "flexDirection": "row",
                            "height": 62,
                            "paddingHorizontal": 10,
                            "width": "100%",
                          }
                        }
                      >
                        <TextInput
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder="Senha"
                          style={
                            {
                              "flex": 1,
                            }
                          }
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  onFocusCapture={[Function]}
                  onLayout={[Function]}
                  style={
                    [
                      {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "width": 702,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "height": 40,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#121212",
                            "fontSize": 24,
                            "fontWeight": "bold",
                          },
                          undefined,
                        ]
                      }
                    >
                      Confirme sua senha
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#222222",
                            "fontSize": 16,
                            "fontWeight": "normal",
                          },
                          undefined,
                        ]
                      }
                    >
                      Digite sua senha novamente
                    </Text>
                    <View
                      style={
                        {
                          "height": 16,
                        }
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "width": "100%",
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "borderColor": "#595959",
                            "borderRadius": 10,
                            "borderWidth": 1,
                            "flexDirection": "row",
                            "height": 62,
                            "paddingHorizontal": 10,
                            "width": "100%",
                          }
                        }
                      >
                        <TextInput
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder="Confirme sua senha"
                          style={
                            {
                              "flex": 1,
                            }
                          }
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#E30B0B",
          "borderRadius": 12,
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 8,
            },
            undefined,
          ]
        }
      >
        Continuar
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
